shell.nix

```diff
@@ -27,18 +27,38 @@ let
   '';
 
   kapply = pkgs.writeShellScriptBin "kapply" ''
+    #!/bin/bash
     set -e
     set -o pipefail
     target="$1"
     shift
     manifests="$(mktemp)"
     kustomize build "$target" > "$manifests"
-    kubectl diff -f "$manifests" "$@"
+    kubectl diff -f "$manifests" "$@" || echo $?
     while true; do
-        read -p "Apply? [y/n] " yn
+        context="$(kubectl config current-context)"
+        read -p "Apply to context $context? [y/n] " yn
         case $yn in
             [Yy]* ) break;;
-            [Nn]* ) exit;;
+            [Nn]* )
+            read -p "Switch context? [y/n] " in
```

Comment:
When changing the context here (assuming that the diff didn't fail due to wrong context), then the diff was run against the wrong context. We really can't approach it this way.